test/gtest/ucp/test_ucp_reconfigure.cc

```diff
@@ -0,0 +1,264 @@
+/**
+* Copyright (c) NVIDIA CORPORATION & AFFILIATES, 2001-2024. ALL RIGHTS RESERVED.
+*
+* See file LICENSE for terms.
+*/
+
+#include "ucp_test.h"
+#include "common/test.h"
+
+extern "C" {
+#include <ucp/core/ucp_ep.inl>
+#include <ucp/wireup/address.h>
+#include <ucp/wireup/wireup_ep.h>
+}
+
+class test_ucp_reconfigure : public ucp_test {
+protected:
+    using address_t = std::pair<void*, ucp_unpacked_address_t>;
+    using address_p = std::unique_ptr<address_t, void (*)(address_t*)>;
+
+    class entity {
+    public:
+        void connect(const entity &other, bool is_exclude_iface);
+
+        void verify_configuration(const entity &other) const;
+
+        bool is_reconfigured() const
+        {
+            return m_cfg_index != ep()->cfg_index;
+        }
+
+        void set_worker_entity(const ucp_test_base::entity *e)
+        {
+            m_worker_entity = e;
+        }
+
+        ucp_ep_h ep() const
+        {
+            return m_ep;
+        }
+
+        ucp_worker_h worker() const
+        {
+            return m_worker_entity->worker();
+        }
+
+        void cleanup()
+        {
+            if (m_ep != nullptr) {
+                ucp_ep_destroy(m_ep);
+            }
+        }
+
+    private:
+        void store_config();
+        ucp_tl_bitmap_t ep_tl_bitmap() const;
+        address_p get_address(bool ep_only) const;
+        bool has_matching_lane(ucp_ep_h ep, ucp_lane_index_t lane_idx,
+                               const entity &other) const;
+
+        ucp_worker_cfg_index_t       m_cfg_index     = UCP_WORKER_CFG_INDEX_NULL;
+        std::vector<uct_ep_h>        m_uct_eps;
+        const ucp_test_base::entity *m_worker_entity = nullptr;
+        ucp_ep_h                     m_ep            = nullptr;
+    };
+
+    void create_entities_and_connect()
+    {
+        m_sender.set_worker_entity(create_entity(true));
+        m_receiver.set_worker_entity(create_entity(false));
+
+        m_sender.connect(m_receiver, is_exclude_iface());
+        m_receiver.connect(m_sender, is_exclude_iface());
+    }
+
+    void cleanup()
+    {
+        m_sender.cleanup();
+        m_receiver.cleanup();
+        ucp_test::cleanup();
+    }
+
+    entity m_sender;
+    entity m_receiver;
+
+public:
+    static void get_test_variants(std::vector<ucp_test_variant> &variants)
+    {
+        add_variant_with_value(variants, UCP_FEATURE_TAG, 0, "");
+        add_variant_with_value(variants, UCP_FEATURE_TAG, 1, "exclude_iface");
+    }
+
+    bool is_exclude_iface() const
+    {
+        return get_variant_value();
+    }
+
+    void send_recv()
+    {
+        static constexpr unsigned num_iterations = 100;
+        static constexpr size_t msg_size         = 16 * UCS_KBYTE;

```

Comment:
why `constexpr`? maybe just `const`?

Reply:
better that it would be initialized at compile time

Reply:
local constant will be optimized the same way by compiler, just replaced with the value calculated in compile time